azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -272,6 +322,15 @@ def _get_telemetry_topic(self):
         topic += "/messages/events/"
         return topic
 
+    def _get_receive_topic(self):
+        topic = "devices/" + self._auth_provider.device_id
+
+        if self._auth_provider.module_id is not None:
+            topic += "/modules/" + self._auth_provider.module_id + "/inputs/#"
+        else:
+            topic += "/messages/devicebound/#"
```

Comment:
You have a hidden assumption here that if you're using a module, then this must be an input message and if you're using a device, then it must be C2D.  This is our current limitation based on the iothub feature set, but I would rather see the implementation be as explicit as possible.  Assuming mutual exclusivity here is not that big a deal, but these kinds of things tend to spiral and pretty soon there's code all over the place that makes teh same assumption.

Reply:
Yeah, I second this - Let's keep distinctions between module and device relegated to the high level code, since high level features can always change. The transport shouldn't know anything about how the upper level code works.

---
In reply to: [246548000](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246548000) [](ancestors = 246548000)

Reply:
sending the topic as a event data now

---
In reply to: [246589578](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246589578) [](ancestors = 246589578,246548000)